torch/optim/sparse_adam.py

```diff
@@ -32,8 +33,12 @@ def __init__(self, params, lr=1e-3, betas=(0.9, 0.999), eps=1e-8):
         if not 0.0 <= betas[1] < 1.0:
             raise ValueError("Invalid beta parameter at index 1: {}".format(betas[1]))
 
+        # if params are in the form of a generator, the next for-loop exhausts it,
+        # so the copy is passed to the loop
+        params, params_copy = itertools.tee(params)
```

Comment:
Is there an advantage to doing this copy as opposed to changing params to a list? Like params = list(params)

Reply:
There is no difference in behavior, but using list instead of itertools.tee seems more readable and easier to understand IMO.